smartsim/_core/launcher/pbs/pbsParser.py

```diff
@@ -83,6 +83,25 @@ def parse_qstat_jobid(output: str, job_id: str) -> str:
     return result
 
 
+def parse_qstat_jobid_json(output: str, job_id: str) -> str:
+    """Parse and return output of the qstat command run with JSON options
+    to obtain job status.
+
+    :param output: output of the qstat command in JSON format
+    :param job_id: allocation id or job step id
+    :return: status
+    """
+    out_json = load_and_clean_json(output)
+
+    if "Jobs" not in out_json:
+        return "NOTFOUND"
+    jobs: dict[str, t.Any] = out_json["Jobs"]
+    job: t.Optional[dict[str, t.Any]] = jobs.get(job_id, None)
+    if job is None:
+        return "NOTFOUND"
```

Comment:
Can we actually make the `"NOTFOUND"` a sentinel value of a different type than string, just so that if someone calls this method they don't accidentally treat the `"NOTFOUND"` sentinel value as a valid job id?

Maybe return `None` when the status cannot be found, and have the return type be a `str | None`?

Reply:
I could up to the calling function. There, when we need to pass it to other functions as argument, it needs to be converted to 'NOTFOUND' as in `result or 'NOTFOUND'` to comply with static typing